src/lib/include/x86_vector_triple.h

```diff
@@ -0,0 +1,51 @@
+#ifndef DISTOPIA_X86_VECTOR_TRIPLE
+#define DISTOPIA_X86_VECTOR_TRIPLE
+
+#include "distopia_type_traits.h"
+#include "x86_swizzle.h"
+#include "x86_vectors.h"
+#include "x86_tgintrin.h"
+
+template <typename VectorT, typename ScalarT, EnableIfVector<VectorT> = 0,
+          EnableIfFloating<ScalarT> = 0, EnableIfMatching<VectorT, ScalarT> = 0>
+class VectorTriple {
+public:
+  VectorT a;
+  VectorT b;
+  VectorT c;
+  // should this be size_t or uintptr_t?
```

Comment:
if we are doing pointer math should this be `uintptr_t` or `std::size_t`

Reply:
`std::size_t` is the convention for sizes. `uintptr_t` is only for when you’re casting from a pointer to an integer